File: 프로그래머스/1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.swift
import Foundation

func solution(_ left:Int, _ right:Int) -> Int {
    var arr: [Int] = Array(repeating: 0, count: 1005)
    var result = 0
    
    for i in 1...right {
        for j in left...right {
            if j.isMultiple(of: i) { arr[j] += 1 }
        }
    }
    
    for i in left...right {
        arr[i].isMultiple(of: 2) ? (result += i) : (result -= i)
    }
    return result
}

File: 프로그래머스/1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.cpp
#include <string>
#include <vector>

using namespace std;

int getFactors(const int& number) {
    int sum = 0;
    for (int i = 1; i * i <= number; i++) {
        if (i * i != number) {
            sum += 2;
        } else {
            sum++;
        }
    }
    return sum;
}

int solution(int left, int right) {
    int answer = 0, cnt = 0;

    for (int i = left; i <= right; i++) {
        cnt = getFactors(i);
        (cnt % 2 == 0) ? answer += i : answer -= i;
    }

    return answer;
}